lib/src/collection/treeset.dart

```diff
@@ -627,10 +625,10 @@ class AvlTreeSet<V> extends TreeSet<V> {
    * See [Set.retainAll]
    */
   void retainAll(Iterable<Object> elements) {
-    List<V> chosen = [];
+    List<V> chosen = <V>[];
     for (var target in elements) {
       if (contains(target)) {
-        chosen.add(target);
+        chosen.add(target as V);
```

Comment:
`contains`, similar to `remove` feeds the item to the comparator, which could get confused if it received a wrong type. What do you think of:

```dart
if (target is V && contains(target)) {
  chosen.add(target as V);
}
```